libs/community/langchain_community/document_loaders/sql_database.py

```diff
@@ -0,0 +1,139 @@
+from typing import Any, Callable, Dict, Iterator, List, Optional, Sequence, Union
+
+import sqlalchemy as sa
+
+from langchain_community.docstore.document import Document
+from langchain_community.document_loaders.base import BaseLoader
+from langchain_community.utilities.sql_database import SQLDatabase
+
+
+class SQLDatabaseLoader(BaseLoader):
+    """
+    Load documents by querying database tables supported by SQLAlchemy.
+
+    For talking to the database, the document loader uses the `SQLDatabase`
+    utility from the LangChain integration toolkit.
+
+    Each document represents one row of the result.
+    """
+
+    def __init__(
+        self,
+        query: Union[str, sa.Select],
+        db: SQLDatabase,
+        *,
+        parameters: Optional[Dict[str, Any]] = None,
```

Comment:
@amotl if you have any cycle documenting how to use this API would be helpful. it's not obvious even after looking at the unit tests, and this will be a blocker for usability

Reply:
Hi @eyurtsev. I am only re-visiting this right now. First of all, thank you very much for taking over from GH-16246, and merging those improvements. We will have a look on the next development iteration how to improve the documentation for this spot you are referring to. Thanks!

Reply:
With "documentation", are you referring to a ..code-block python example, like mentioned over at https://github.com/langchain-ai/langchain/pull/16246#discussion_r1487139999?